crypt_test.go

```diff
@@ -2524,7 +2547,7 @@ func (s *cryptSuite) TestInitializeLUKS2ContainerWithCustomKDFIterations(c *C) {
 }
 
 func (s *cryptSuite) TestInitializeLUKS2ContainerInvalidKeySize(c *C) {
-	c.Check(InitializeLUKS2Container("/dev/sda1", "data", s.newPrimaryKey()[0:16], nil), ErrorMatches, "expected a key length of at least 256-bits \\(got 128\\)")
+	c.Check(InitializeLUKS2Container("/dev/sda1", "data", ([]byte)(s.keyDataTestBase.newPrimaryKey(c, 32))[0:16], nil), ErrorMatches, "expected a key length of at least 256-bits \\(got 128\\)")
```

Comment:
As `keyDataTestBase` takes a size now, you don't need to create a 32-byte key here and then truncate it to 16 bytes.

Also, is the type conversion to `[]byte` necessary here? That should happen implicitly because `[]byte` is not a named type. This is repeated in a few other places too.

For an idea of when explicit conversions are necessary when converting between types, take a look at this section of the language spec: [Assignability](https://go.dev/ref/spec#Assignability) (in this case, "V and T have identical [underlying types](https://go.dev/ref/spec#Underlying_types) but are not type parameters and at least one of V or T is not a [named type](https://go.dev/ref/spec#Types)).

Reply:
Removed the truncation, thanks. 

Regarding the type conversion, keyDataTestBase`s `newPrimaryKey` returns type `PrimaryKey` while all the LUKS related APIs expect type `DiskUnlockKey` so I believe this doesn't fall under the categories above so the type conversion is needed despite them having the same underlying type `[]byte`.

Reply:
Ah, I think the problem here is that perhaps there should have originally been a `newUnlockKey` method for returning a key with the correct type.